pipeline/train/train.py

```diff
@@ -0,0 +1,413 @@
+
+        for alignment_file in self.alignments_files:
+            if not alignment_file.exists():
+                raise Exception(f"Alignment file could not be found {alignment_file}")
+
+    def build_datasets(self) -> None:
+        # Start by building the training datasets, e.g.
+        #
+        #  corpus.enfr.tsv from:
+        #   - fetches/corpus.en.zst
+        #   - fetches/corpus.fr.zst
+        #   - fetches/corpus.aln.zst
+        #
+        #  mono.enfr.tsv from:
+        #   - fetches/mono.en.zst
+        #   - fetches/mono.fr.zst
+        #   - fetches/mono.aln.zst
+
+        for index, dataset_prefix in enumerate(self.train_set_prefixes):
+            alignments = None
+            if self.alignments_files:
+                alignments = self.alignments_files[index]
+                self.config_variables[f"dataset{index}"] = build_dataset_tsv(
+                    dataset_prefix, self.src, self.trg, alignments
+                )
+            else:
+                self.config_variables[f"dataset{index}"] = build_dataset_tsv(
+                    dataset_prefix, self.src, self.trg
+                )
+
+        # Then build out the validation set, for instance:
+        #
+        # devset.enfr.tsv from:
+        #  - fetches/devset.en.zst
+        #  - fetches/devset.fr.zst
+        self.validation_set = build_dataset_tsv(self.validation_set_prefix, self.src, self.trg)
+
+    def generate_opustrainer_config(
+        self,
+    ):
+        """
+        Generate an OpusTraininer config that points to the current datasets and language
+        options.
+        """
+        if self.model_type == ModelType.teacher:
+            teacher_mode = self.teacher_mode.value
+            if teacher_mode == TeacherMode.none.value:
+                logger.error("Teacher mode was not properly set")
+                teacher_mode = TeacherMode.two_stage.value
+
+            config_input = (
+                train_dir / f"configs/opustrainer/{self.model_type.value}.{teacher_mode}.yml"
+            )
+        else:
+            config_input = train_dir / f"configs/opustrainer/{self.model_type.value}.yml"
+
+        with open(config_input, "rt", encoding="utf-8") as file:
+            config_text = file.read()
+
+        logger.info(f"Applying OpusTrainer config variables: {config_input}")
+        for key, value in self.config_variables.items():
+            logger.info(f" - {key}: {value}")
+
+        with self.opustrainer_config.open("wt", encoding="utf-8") as file:
+            config_text = config_text.format(**self.config_variables)
+            file.write(config_text)
+
+    def get_opustrainer_cmd(self):
+        return [
+            "opustrainer-train",
+            *apply_command_args(
+                {
+                    "config": str(self.opustrainer_config),
+                    "log-file": str(self.artifacts / "opustrainer.log"),
+                    "log-level": "ERROR",
```

Comment:
Log level should be switched to info if we make sure that there are no empty alignments. See https://github.com/mozilla/firefox-translations-training/issues/617

Reply:
Ah, I was going to follow-up with that, but happy to do it here.